service/controller/v5/blobclient/error.go

```diff
@@ -0,0 +1,86 @@
+package blobclient
+
+import (
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/giantswarm/microerror"
+	"strings"
+)
+
+var executionFailedError = &microerror.Error{
+	Kind: "executionFailedError",
+}
+
+// IsExecutionFailed asserts executionFailedError.
+func IsExecutionFailed(err error) bool {
+	return microerror.Cause(err) == executionFailedError
+}
+
+var invalidConfigError = &microerror.Error{
+	Kind: "invalidConfigError",
+}
+
+// IsInvalidConfig asserts invalidConfigError.
+func IsInvalidConfig(err error) bool {
+	return microerror.Cause(err) == invalidConfigError
+}
+
+var notFoundError = &microerror.Error{
+	Kind: "notFoundError",
+}
+
+// IsNotFound asserts notFoundError.
+func IsNotFound(err error) bool {
+	if err == nil {
+		return false
+	}
+
+	c := microerror.Cause(err)
+
+	if c == notFoundError {
+		return true
+	}
+
+	{
+		dErr, ok := c.(autorest.DetailedError)
+		if ok {
+			if dErr.StatusCode == 404 {
+				return true
+			}
+		}
+	}
+
+	return false
+}
+
+// IsBlobNotFound asserts blob not found error from upstream's API message.
+func IsBlobNotFound(err error) bool {
+	if err == nil {
+		return false
+	}
+	return strings.Contains(microerror.Cause(err).Error(), "ServiceCode=BlobNotFound")
+}
+
+// IsContainerNotFound asserts container not found error from upstream's API message.
+func IsContainerNotFound(err error) bool {
```

Comment:
There are matchers that do not have an associated error type. Let's please stay consistent and have them.

Reply:
By staying consistent do you mean just having them without usage? Those are an error which in my case do not go into the stack but just wait for next loop iteration, because of resources itself are created with arm templates

Reply:
We had this discussion already. IMO having a not used error variable is confusing.

Reply:
The scheme you scan without needing to think is usually error, matcher, error, matcher, and so on. When this changes you have to start to think. That takes time. That takes energy. It is the same for alphabetical order. We are as efficient as we are because we have a couple of consistency standards and all of them make it pretty easy and fast to work with a project even if you are not used to it on a daily basis. Errors and matchers come in pairs. Let's stick to the guidelines we give ourselves. When you want to change the guideline, get consensus on them first.

Reply:
I didn't see a guideline saying having an error variable for the matcher is mandatory. So I don't think there is any consensus on that. IMO having a not used error variable is confusing. I won't fight for that though. It's a minor thing for me.

Reply:
All the examples show always both. See https://github.com/giantswarm/fmt/blob/master/go/errors.md.